common/common-ui/src/main/java/com/duckduckgo/common/ui/experiments/visual/store/VisualDesignExperimentDataStoreImpl.kt

```diff
@@ -16,149 +16,48 @@
-        prefsKey = KEY_NAVIGATION_BAR_ENABLED,
-        prefsDefault = true, // for now, navigation bar always enabled when the whole experiment enabled
-    )
-    override val navigationBarState: StateFlow<FeatureState> = _navigationBarState.state
-
-    init {
-        updateFeatureState()
-    }
+    override val isExperimentEnabled: StateFlow<Boolean> = _experimentFeatureFlagEnabled.asStateFlow()
 
     override fun onPrivacyConfigDownloaded() {
         updateFeatureState()
     }
 
-    override fun setExperimentStateUserPreference(enabled: Boolean) {
-        _experimentState.setUserPreference(enabled)
+    override fun changeExperimentFlagPreference(enabled: Boolean) {
+        experimentalUIThemingFeature.self().setRawStoredState(Toggle.State(remoteEnableState = enabled))
+        updateFeatureState()
     }
 
     private fun updateFeatureState() {
-        _experimentState.updateFeatureState()
-        _navigationBarState.updateFeatureState()
-    }
-
-    private class ExperimentFeature(
-        private val store: DataStore<Preferences>,
-        private val coroutineScope: CoroutineScope,
-        parentStateFlow: Flow<FeatureState> = NO_PARENT_FEATURE,
-        private val targetToggle: Toggle,
-        private val prefsKey: String,
-        private val prefsDefault: Boolean,
-    ) {
-        private val _available = MutableStateFlow(targetToggle.isEnabled())
-        private val _enabled = store.data.map { preferences ->
-            preferences[booleanPreferencesKey(prefsKey)] ?: prefsDefault
-        }
-
-        /**
-         * Returns [FeatureState] of this feature.
-         *
-         * If the designated parent feature is disabled, both [FeatureState.isAvailable] and [FeatureState.isEnabled] will report `false`,
-         * until the parent feature gets enabled.
-         */
-        val state: StateFlow<FeatureState> = combine(
-            _available,
-            _enabled,
-            parentStateFlow,
-        ) { available, userPrefEnabled, parentState ->
-            FeatureState(
-                isAvailable = parentState.isEnabled && available,
-                isEnabled = parentState.isEnabled && available && userPrefEnabled,
-            )
-        }.stateIn(
-            coroutineScope,
-            SharingStarted.Eagerly,
-            initialValue = FeatureState(
-                isAvailable = _available.value,
-                // FIXME Synchronously fetching the initial preference value is wrong but it's a temporary workaround
-                //  because the omnibar implementation that relies on this value can't recreate itself dynamically,
-                //  it requires the whole activity to be recreated. To avoid that recreation right after initial launch,
-                //  we need the initial value to actually use the valid persisted state.
-                isEnabled = _available.value && runBlocking { _enabled.firstOrNull() == true },
```

Comment:
We're removing the synchronous check for the data store, only a single synchronous check for the feature toggle remains. However, by the time the feature toggle is accessed, the config should already be cached, so this should in theory resolve the minor ANR issues we've been seeing.